src/Sentry.Unity/UnityEventProcessor.cs

```diff
@@ -117,9 +117,7 @@ private void PopulateDevice(Device device)
                 device.BatteryStatus = SystemInfo.batteryStatus.ToString(); // don't cache
 
                 var batteryLevel = SystemInfo.batteryLevel;
-#pragma warning disable RECS0018 // Value is exact when expressing no battery level
-                if (batteryLevel != -1.0)
-#pragma warning restore RECS0018
+                if (batteryLevel > 0.0)
```

Comment:
IIRC this was due to some iOS device reporting the battery status as `-1`.

Reply:
which is lower than 0.0 so the new code would also catch that. It's just that comparing exact floating values doesn't work properly, as the disabled warning was saying.

IMO, no need to try and report negative battery level - it's not a valid value.